gatsby-config.js

```diff
@@ -8,198 +10,211 @@ require("dotenv").config({
   path: `.env.${process.env.NODE_ENV}`,
 })
 
-// Gatsby Configuration, Options, and Plugins
-module.exports = {
-  // Puts build artifacts in a subdirectory, and updates all local links
-  pathPrefix: `/docs`,
-  // Reusable global information
-  siteMetadata: {
-    title: `Pantheon Docs`,
-    description: `Information for building, launching, and running dynamic sites on the Pantheon Website Management Platform`,
-    siteUrl: `https://pantheon.io/`,
-    social: {
-      twitter: `getpantheon`,
-    },
+const plugins = [
```

Comment:
I don't think it's necessary to create a const to handle this functionality, we can conditionally add the element to the array 

```
{
    ...(some_validations && {
      resolve: `gatsby-plugin-algolia`, 
      options: {
        appId: config.search.algoliaAppId, // algolia application id
        apiKey: config.search.algoliaAdminKey, // algolia admin key
        queries,
        chunkSize: 10000, // default: 1000
      }
    })
  }
```